### PR TITLE
Pick only user settings connections for OE and its features

### PR DESCRIPTION
**Pull request by abist**
Fixes https://github.com/microsoft/vscode-mssql/issues/1353

Interact OE with only user settings connections and keep MRU away from OE.

### CODE REVIEW

**Comment by abist** [comment]
@kburtram @Charles-Gagnon gentle ping :)

File: src/models/connectionStore.ts [hunk 1]
```diff
@@ -391,11 +391,14 @@ export class ConnectionStore {
     }
 
     // Load connections from user preferences
-    public loadAllConnections(): IConnectionCredentialsQuickPickItem[] {
+    public loadAllConnections(addRecentConnections: boolean = true): IConnectionCredentialsQuickPickItem[] {
```
Comment (kburtram) [inline src/models/connectionStore.ts]: it seems like all invocations of this method are calling with false.  Maybe we should make the default false?

**Comment by kburtram** [comment]
@abist regarding the "ping", please keep in mind that the code review SLA we're targeting is ~24 hour turn around time.  Expecting a review within a couple hours of posting a PR may ending up being a bit much.  Though it doesn't hurt to ask I suppose 😄.

File: src/models/connectionStore.ts (hunk 1 above)
Comment (abist) [reply src/models/connectionStore.ts]: Only in the case of OE features. For queries, people still might use the MRU.

**Pull request merged**